File: Java/1863.java
class Solution {
    public int subsetXORSum(int[] nums) {
        return helper(nums, 0, 0);
    }

    private int helper(int[] nums, int i, int currentResult) {
        if (i == nums.length) {
            return currentResult;
        }
        return helper(nums, i + 1, currentResult ^ nums[i]) + helper(nums, i + 1, currentResult);
    }
}

File: Python/1863.py
from typing import List


class Solution:
    def subsetXORSum(self, nums: List[int]) -> int:

        def backtracking(i, curr_xor):
            if i == len(nums):
                return curr_xor

            include = backtracking(i + 1, curr_xor ^ nums[i])
            exclude = backtracking(i + 1, curr_xor)

            return include + exclude

        return backtracking(0, 0)
